defaults.properties.yml

```diff
@@ -1,30 +1,192 @@
 
+  # Name of the sites subdirectory where the `settings.php` file should live.
+  sites_subdir: default
+
+  # Path to the Drupal config directory, relative to the Drupal root.
+  config_sync_directory: "../config/sites/${drupal.sites_subdir}"
+
+  # Drupal admin username, if you feel inclined to change it.
+  admin_user: admin
+
+  # Whether to allow running the drupal-install target when there are local changes to the
+  # exported Drupal config.
+  allow_dirty_config: no
+
+  # Database connection for Drupal.
   database:
     database: drupal
     username: root
     password: root
     host: 127.0.0.1
 
   settings:
+    # Path to the Drupal public files directory, relative to the Drupal root.
     file_public_path: sites/default/files
+
+    # Path to the Drupal private files directory, relative to the Drupal root.
     file_private_path: ""
 
   twig:
+    # Whether to enable twig debugging.
     debug: false
 
+# Build configuration used by tasks/drupal.xml
+build:
+  drupal:
+    # Source template for Drupal's `settings.php` file. This may vary per build
+    # environment.
+    settings: .the-build/drupal.settings.build.php
+
+    # Destination for the templated settings.php file.
+    settings_dest: "${drupal.root}/sites/${drupal.sites_subdir}/settings.build.php"
+
+    # Source template for Drupal's `services.yml` file. This may vary per build
+    # environment.
+    services: .the-build/drupal.services.build.yml
+
+    # Destination for the templated `services.yml` file.
+    services_dest: "${drupal.root}/sites/${drupal.sites_subdir}/services.build.yml"
+
+# Configuration for tasks/artifact.xml
 artifact:
+  # The path of the working directory where the artifact should be built.
   directory: artifacts/build
+
+  # A string prefix to use for branch names.
+  prefix: artifact-
+
+  # Path to a template .gitignore file to use in the artifact.
+  gitignore_template: "${build.thebuild.dir}/defaults/templates/artifact.gitignore"
+
+  # Path to a template README file to use in the artifact.
+  readme_template: "${build.thebuild.dir}/defaults/templates/artifact.README.md"
+
   git:
-    # The following property is commented out, because this value should only be set in
-    # per-project properties files.
-    # remote:
-      
+    # Git repository for the artifact. This is typically an Acquia or Pantheon git URL.
+    # This property is commented out, because the value should only be set in per-project
+    # properties files.
+    #remote:
+
+    # Name of the remote branch to push the artifact to.
+    # The remote_branch property is generally set dynamically, to 'artifact.prefix' plus
+    # the current branch name. However, it may be overridden either in your project
+    # properties or in a run-time flag (-Dartifact.git.remote_branch=foo) in order to
+    # build to a static destination branch, or a branch name that matches other
+    # constraints (like Pantheon's 11 character limit for multidev environment branch
```

Comment:
Great example!

Reply:
This was totally from my work with Avi on NTC -- we were so confused until we figured out Pantheon's branch naming restrictions :(

Reply:
👍